### PR TITLE
Normalize `<non_null_ref> is null` to `false`

### PR DESCRIPTION
**Pull request by jeeminso**
## Summary of the changes / Why this improves CrateDB
Fixes https://github.com/crate/crate/issues/15395

The original bug scenario fails because `EqualityExtractor` could not handle `WHERE NOT ((t1.c0 != '1' )AND (t1.c0 IS NULL));` properly. But instead of fixing `EqualityExtractor` I took the approach to normalize `<non_null_ref> is null` to `false`.

## Checklist

 - [x] Added an entry in the latest `docs/appendices/release-notes/<x.y.0>.rst` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in the latest `docs/appendices/release-notes/<x.y.0>.rst`
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: server/src/main/java/io/crate/analyze/where/EqualityExtractor.java
Comment (jeeminso) [inline server/src/main/java/io/crate/analyze/where/EqualityExtractor.java:142]: If handled this way, it becomes difficult to identify either `a=1 <op> a is null` or `<op>(a=1 or a is null)` or both `<op>`.

**Comment by jeeminso** [comment]
Another approach: rewrite `<symbol> is null` into `true` or `false` if possible. For pk columns that are `non null` by default, `<pk_col> is null` can be rewritten into `false`.

File: server/src/main/java/io/crate/analyze/where/EqualityExtractor.java
Comment (jeeminso) [reply server/src/main/java/io/crate/analyze/where/EqualityExtractor.java:142]: Not taking this approach.

File: server/src/main/java/io/crate/expression/predicate/IsNullPredicate.java
Comment (jeeminso) [inline server/src/main/java/io/crate/expression/predicate/IsNullPredicate.java:95]: Testcases like `testSelectWhereVersionIsNullPredicate()` fails because `WhereClauseValidator` happens _after_ the non-null ref is evaluated.

File: server/src/main/java/io/crate/expression/predicate/IsNullPredicate.java
Comment (jeeminso) [inline server/src/main/java/io/crate/expression/predicate/IsNullPredicate.java:98]: This normalization is limited to columns only, need to expand it to handle expressions, ex) `<not_null_ref> + 1 is null`.
This also reminds me of https://github.com/crate/crate/blob/efeb10240fe14e11141b82019e2e9d381c4e3da5/server/src/main/java/io/crate/expression/predicate/NotPredicate.java#L184-L196 which also does something similar(but for `Collect` operation not `Get` as in the original bug report).
Once the proper normalization is in place, it should be able to be removed.

File: docs/appendices/release-notes/5.6.1.rst
Comment (matriv) [inline docs/appendices/release-notes/5.6.1.rst:58]:
```suggestion
- Fixed an issue that caused queries to return invalid results when the
```

File: server/src/main/java/io/crate/expression/predicate/IsNullPredicate.java
Comment (matriv) [inline server/src/main/java/io/crate/expression/predicate/IsNullPredicate.java:94]: thx for the comment, is there an easy way to validate the usage of `_version` before we do the normalization? I see for example in `RelationAnalyzer#visitQuerySpecification()` that the `WhereClauseValidator` is called before the symbol normalization, but maybe there are other cases?

File: server/src/test/java/io/crate/planner/WhereClauseOptimizerTest.java
Comment (matriv) [inline server/src/test/java/io/crate/planner/WhereClauseOptimizerTest.java:223]: This doesn't relate to the bug fix, right?
Could you please double check that we don't test this functionality elsewhere?

File: server/src/test/java/io/crate/planner/WhereClauseOptimizerTest.java
Comment (matriv) [inline server/src/test/java/io/crate/planner/WhereClauseOptimizerTest.java:235]: same

**Review by matriv — COMMENTED** [review]
thx! looks good, left some questions/comments.

File: server/src/test/java/io/crate/planner/WhereClauseOptimizerTest.java
Comment (jeeminso) [reply server/src/test/java/io/crate/planner/WhereClauseOptimizerTest.java:223]: Thank you,

porting these tests from `AndOperatorTest` to `OrOperatorTest` then removed the ones with `or true`/`and false`:
```
    @Test
    public void testNormalizeBooleanTrueAndNonLiteral() throws Exception {
        assertNormalize("is_awesome and true", isReference("is_awesome"));
    }

    @Test
    public void testNormalizeBooleanFalseAndNonLiteral() throws Exception {
        assertNormalize("is_awesome and false", isLiteral(false));
    }
```

File: server/src/test/java/io/crate/expression/operator/OrOperatorTest.java
Comment (matriv) [inline server/src/test/java/io/crate/expression/operator/OrOperatorTest.java:34]: Just a naming suggestion, but your call if you prefer
```suggestion
    public void test_normalize_boolean_true_or_reference() throws Exception {
```

or 

```suggestion
    public void test_normalize_boolean_true_or_column() throws Exception {
```

File: server/src/test/java/io/crate/expression/operator/OrOperatorTest.java
Comment (matriv) [inline server/src/test/java/io/crate/expression/operator/OrOperatorTest.java:39]: same

**Review by matriv — APPROVED** [review]
thx!

**Comment by matriv** [comment]
@jeeminso Before merging, please add release entry (and the file `5.5.4.rst`, which is not currently there) for 5.5.4.

**Comment by BaurzhanSakhariev** [comment]
>and the file `5.5.4.rst`, which is not currently there

https://github.com/crate/crate/pull/15397 contains it and will be merged soon

**Pull request merged**
